### PR TITLE
chore: Added docs for payment connector create

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Enhancement

## Description

API spec for Payment Connector API

## Motivation and Context

N/A

## How did you test it?

Manual

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/api_models/src/admin.rs [hunk 1]
```diff
@@ -163,33 +163,101 @@ pub struct MerchantConnectorId {
...
+/// Create a new Payment Connector for the merchant account. The connector could be a payment processor / facilitator / acquirer or specialized services like Fraud / Accounting etc."
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
 #[serde(deny_unknown_fields)]
 pub struct PaymentConnectorCreate {
+    /// Type of the Connector for the financial use case. Could range from Payments to Accounting to Banking.
+    #[schema(example = "payment_processor,payment_vas")]
     pub connector_type: api_enums::ConnectorType,
+    /// Name of the Connector
+    #[schema(value_type = String,example = "stripe,adyen")]
     pub connector_name: String,
```
Comment (SanchithHegde) [inline crates/api_models/src/admin.rs:176]: 1. The examples here would make it seem that the field accepts a comma-separated list of values, which is not the case. Please include only one value as an example for these fields.
2. Explicitly specifying `value_type` for `String` is not required.

File: crates/api_models/src/admin.rs [hunk 2]
```diff
@@ -163,33 +163,101 @@ pub struct MerchantConnectorId {
...
+    /// Unique ID of the connector
+    #[schema(value_type = Option<i32>,example = 42)]
     pub merchant_connector_id: Option<i32>,
```
Comment (SanchithHegde) [inline crates/api_models/src/admin.rs:179]: Explicitly specifying `value_type` for `Option<i32>` is not required.

File: crates/api_models/src/admin.rs [hunk 3]
```diff
@@ -163,33 +163,101 @@ pub struct MerchantConnectorId {
...
+    #[schema(value_type = Option<Object>,example = json!({ "auth_type": "HeaderKey","api_key": "Basic MyVerySecretApiKey" }))]
     pub connector_account_details: Option<Secret<serde_json::Value>>,
+    /// A boolean value to indicate if the connector is in Test mode. By default, its value is false.
+    #[schema(value_type = Option<bool>,default = false, example = false)]
     pub test_mode: Option<bool>,
+    /// A boolean value to indicate if the connector is disabled. By default, its value is false.
+    #[schema(value_type = Option<bool>,default = false, example = false)]
     pub disabled: Option<bool>,
+    /// Refers to the Parent Merchant ID if the merchant being created is a sub-merchant
+    #[schema(value_type = Option<Vec<PaymentMethods>>,example = json!([
```
Comment (SanchithHegde) [inline crates/api_models/src/admin.rs:190]: Explicitly specifying `value_type` for `Option<bool>` and `Option<Vec<PaymentMethods>>` is not required.

File: openapi/generated.json [hunk 4]
```diff
@@ -320,13 +322,7 @@
...
           },
           "routing_algorithm": {
-            "$ref": "#/components/schemas/RoutingAlgorithm"
-          },
-          "custom_routing_rules": {
-            "type": "array",
-            "items": {
-              "$ref": "#/components/schemas/CustomRoutingRules"
-            }
+            "$ref": "#/components/schemas/serde_json.Value"
```
Comment (SanchithHegde) [inline openapi/generated.json:325]: Please update `value_type` for `routing_algorithm` in `CreateMerchantAccount` struct to be `Option<Object>`.

File: openapi/generated.json [hunk 5]
```diff
@@ -1074,6 +896,94 @@
...
+      "PaymentConnectorCreate": {
+        "type": "object",
+        "description": "Create a new Payment Connector for the merchant account. The connector could be a payment processor / facilitator / acquirer or specialized services like Fraud / Accounting etc.\"",
+        "required": [
+          "connector_type",
+          "connector_name"
+        ],
+        "properties": {
+          "connector_type": {
+            "$ref": "#/components/schemas/api_enums.ConnectorType"
```
Comment (SanchithHegde) [inline openapi/generated.json:908]: Specify the `value_type` for this field as `ConnectorType`.

File: crates/api_models/src/admin.rs [hunk 6]
```diff
@@ -163,33 +163,101 @@ pub struct MerchantConnectorId {
...
     pub payment_method_issuers: Option<Vec<String>>,
+    /// List of payment schemes accepted or has the processing capabilities of the processor
+    #[schema(value_type = Option<Vec<String>>,example = "[MASTER,VISA,DINERS]")]
     pub payment_schemes: Option<Vec<String>>,
+    /// List of currencies accepted or has the processing capabilities of the processor
+    #[schema(value_type = Option<Vec<Currency>>,example = "[USD,EUR,AED]")]
     pub accepted_currencies: Option<Vec<api_enums::Currency>>,
+    ///  List of Countries accepted or has the processing capabilities of the processor
+    #[schema(value_type = Option<Vec<String>>,example = "[US,IN]")]
     pub accepted_countries: Option<Vec<String>>,
```
Comment (SanchithHegde) [inline crates/api_models/src/admin.rs:245]: Can you use `json!(["item1", "item2"])` for specifying the examples here?

File: crates/api_models/src/admin.rs [hunk 7]
```diff
@@ -163,33 +163,101 @@ pub struct MerchantConnectorId {
...
+    #[schema(value_type = Option<i32>,example = 1313)]
     pub maximum_amount: Option<i32>,
+    /// Boolean to enable recurring payments / mandates. Default is true.
+    #[schema(default = true, value_type = bool,example = false)]
     pub recurring_enabled: bool,
+    /// Boolean to enable installment / EMI / BNPL payments. Default is true.
+    #[schema(default = true, value_type = bool,example = false)]
     pub installment_payment_enabled: bool,
+    /// Type of payment experience enabled with the connector
+    #[schema(value_type = Option<Vec<PaymentExperience>>,example = "[redirect_to_url]")]
```
Comment (SanchithHegde) [inline crates/api_models/src/admin.rs:260]: Same here, use `json!(["..."])` instead.

**Review by SanchithHegde — CHANGES_REQUESTED** [review]
Other than that, looks good to me!

File: crates/api_models/src/admin.rs [hunk 8]
```diff
@@ -33,6 +33,7 @@ pub struct CreateMerchantAccount {
     pub webhook_details: Option<WebhookDetails>,
 
     /// The routing algorithm to be used for routing payments to desired connectors
+    #[schema(value_type = Option<Object>,example = "round_robin")]
     pub routing_algorithm: Option<serde_json::Value>,
```
Comment (SanchithHegde) [inline crates/api_models/src/admin.rs:37]:
```suggestion
    #[schema(value_type = Option<Object>,example = json!({"type": "single", "data": "stripe"})]
    pub routing_algorithm: Option<serde_json::Value>,
```

**Review by SanchithHegde — CHANGES_REQUESTED** [review]
Other than that, looks good to me!

**Review by SanchithHegde — DISMISSED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**